# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/localstorage/ActorsParent.cpp
# Commit: 5ae0cf36c38b
# Full Hash: 5ae0cf36c38bc5d79553382de16f41a2795de86c
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-13 21:18:59
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add support for initializing corresponding origin to QuotaManager::OpenClientDirectory; r=devtools-reviewers,nchevobbe
#   
#   QuotaManager::OpenClientDirectory can now get initialization status of
#   corresponding origin and initialize the origin if it's needed.
#   
# ==============================================================================

diff -r 7b3ae7465c92 -r 5ae0cf36c38b dom/localstorage/ActorsParent.cpp
--- a/dom/localstorage/ActorsParent.cpp	Sun Oct 13 10:10:23 2024 +0000
+++ b/dom/localstorage/ActorsParent.cpp	Sun Oct 13 10:10:23 2024 +0000
@@ -6799,6 +6799,7 @@
 
   quotaManager
       ->OpenClientDirectory({mOriginMetadata, mozilla::dom::quota::Client::LS},
+                            /* aCreateIfNonExistent */ false,
                             SomeRef(mPendingDirectoryLock))
       ->Then(
           GetCurrentSerialEventTarget(), __func__,